static/web.js

```diff
@@ -241,48 +293,63 @@
         }, button, "Creating Gist…", result);
     }
 
-    function httpRequest(method, url, data, expect, on_success, on_fail) {
-        var req = new XMLHttpRequest();
-
-        req.open(method, url, true);
-        req.onreadystatechange = function () {
-            if (req.readyState == XMLHttpRequest.DONE) {
-                if (req.status == expect) {
-                    if (on_success) {
-                        on_success(req.responseText);
-                    }
-                } else {
-                    if (on_fail) {
-                        on_fail(req.status, req.responseText);
-                    }
-                }
-            }
+    /**
+     * 
+     * @param {'GET'|'HEAD'|'POST'|'PUT'|'DELETE'|'CONNECT'|'OPTIONS'|'TRACE'|'PATCH'} method (see [HTTP request methods](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods))
+     * @param {String} url 
+     * @param {any} data 
+     * @param {Number} expect 
+     * @param {Function} on_success 
+     * @param {Function} on_fail 
+     * @return {void}
+     */
+    async function httpRequest(method, url, data, expect, on_success, on_fail) {
+        const options = {
+            method,          
         };
-
-        if (method === "GET") {
-            req.send();
-        } else if (method === "POST") {
-            req.send(data);
+        if (method === 'POST') {
+            options.body = data;
         }
-    }
-
-    function fetchGist(session, result, gist_id, do_evaluate, evaluateButton) {
+        const response = await fetch(url, options);
+        //const responseText = await response.text()
```

Comment:
Can this commented line be removed?

Reply:
I'll keep that in mind. I only did this because these were so many minor changes and I didn't want to float your PRs with like 20 of them.

> Can this commented line be removed?

Sure thing. I'll do that.